facebook/ThreatExchange
# [UI] Refactor Banks Page

Summary
---------

This PR refactors the bank grid page, match form, create bank form, and add content form

- the banks are now listed in a table instead of a grid of cards
- The Match Form is now in a card with different styling
- the Add new bank form is now in a modal
- the add content form is now in a modal that is triggered by a button per item in the table (there is also now a success state for it)


Test Plan

- create a new bank and verify that it adds like expected
- add content to banks and verify that you see a success state as expected with a hash value returned
- Search by that same content and verify that you see the matches you expect (matches will highlight in the table)

---------




Screenshots:

<img width="1511" alt="Screenshot 2024-10-04 at 9 19 32 AM" src="https://github.com/user-attachments/assets/3b20fc09-e945-4cbe-9d23-fa71e4f45eb4">
<img width="1511" alt="Screenshot 2024-10-04 at 9 09 38 AM" src="https://github.com/user-attachments/assets/d978dfc9-ad92-4754-ae0c-bb846fe07e9d">
<img width="1512" alt="Screenshot 2024-10-04 at 9 10 05 AM" src="https://github.com/user-attachments/assets/cfce9c59-a2a4-4d93-8c8d-4a39cd0b4e90">
<img width="1512" alt="Screenshot 2024-10-04 at 9 10 19 AM" src="https://github.com/user-attachments/assets/b5af9e1b-2530-4b07-b813-4e559e1a99b9">
<img width="1512" alt="Screenshot 2024-10-04 at 9 10 33 AM" src="https://github.com/user-attachments/assets/d1aab9e3-9a95-42b9-a7fa-362dcdcfb84c">
